### PR TITLE
Move completion files in data, and install the bash-completion file

### LINKED ISSUE #1101: Install yarp_completion file
It would be a helpful inclusion for an installation from precompiled binaries.

### PR DESCRIPTION
**Pull request by drdanz**
:warning: Warning: the WindowsPowerShell and the bash completion files are now in a different directories, some scripts might require to be updated.

* The `scripts/WindowsPowerShell` files are now in `data/WindowsPowerShell`
* The `scripts/yarp_completion` file is now in `data/bash-completion/yarp`
* The `bash-completion` file is now installed automatically (#1101)

Fixes #1101

### CODE REVIEW

**Comment by pattacini** [comment]
@drdanz does it work for both Linux and Windows platforms at `make install` time?

**Comment by drdanz** [comment]
No, sorry, the WindowsPowerShell installation is more complicated. The bash completion files are installed anyway on windows though, I don't see a big issue in having this file installed on all systems.

File: data/bash-completion/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Copyright (C) 2006-2020 Istituto Italiano di Tecnologia (IIT)
+# All rights reserved.
+#
+# This software may be modified and distributed under the terms of the
+# BSD-3-Clause license. See the accompanying LICENSE file for details.
+
+yarp_install(FILES yarp
+             DESTINATION ${CMAKE_INSTALL_DATADIR}/bash-completion/completion
```
Comment (pattacini) [inline data/bash-completion/CMakeLists.txt]: @drdanz 

According to this [post][1], we have:
```sh
$ pkg-config --variable=completionsdir bash-completion
/usr/share/bash-completion/completions
$
$ pkg-config --variable=compatdir bash-completion
/etc/bash_completion.d
```

Therefore:
- it seems that the location where yarp completion gets copied into should end with `completions` and not `completion`
- anyway, that location is not automatically recognized by the system; thus, is there any way to include it?


[1]: https://serverfault.com/a/506707

**Pull request merged**